### ISSUE TITLE
Kubernetes 1.6 showing DNS errors on deployments

### ISSUE THREAD

**Issue by ryanwalls** [rancher/rancher#9303]
**Rancher versions:**
rancher/server: 1.6.3
rancher/agent: rancher/agent:v1.2.5

**Infrastructure Stack versions:**
healthcheck: 0.3.0
ipsec: 0.1.1
network-services: 0.2.2
scheduler: rancher/k8s:v1.6.6-rancher1-4
kubernetes (if applicable): rancher/k8s:v1.6.6-rancher1-4

**Docker version: (`docker version`,`docker info` preferred)**
```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:38:45 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:38:45 2017
 OS/Arch:      linux/amd64
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
ubuntu@ip-172-31-11-231:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.2 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.2 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
ubuntu@ip-172-31-11-231:~$ uname -r
4.4.0-1013-aws
ubuntu@ip-172-31-11-231:~$ 
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher with external DB

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes

**Steps to Reproduce:**
Upgrade from 1.5.4 k8s to 1.6.6.
Scale a deployment up by 1 pod.  

**Results:**
Deployment prints this error: 
```
Search Line limits were exceeded, some dns names have been omitted, the applied search line is: default.svc.cluster.local svc.cluster.local cluster.local kubelet.kubernetes.rancher.internal kubernetes.rancher.internal rancher.internal
```

![image](https://user-images.githubusercontent.com/842653/28071620-af2a38f0-660d-11e7-9e17-115071411b33.png)

**Comment by FabienD** [comment]
Hi, 

I've got exactly the same event with a similar stack runs on a different OS (CoreOS 1409.6.0). When the deployment is finished, the event goes away. I didn't notice the same error under k8s 1.5.4.

Fabd.

**Comment by obriensystems** [comment]
temporarily fixed by adjusting /etc/resolve.conf

Fix DNS resolution before running any more pods ( add service.ns.svc.cluster.local)
root@obriensystemskub0:~/oom/kubernetes/oneclick# cat /etc/resolv.conf
nameserver 192.168.241.2
search localdomain service.ns.svc.cluster.local

**Comment by superseb** [comment]
The log line was added in 1.6. Can you confirm that you're not experiencing interruptions or errors due to this message? (because it shouldn't do any harm).

**Comment by ryanwalls** [comment]
@superseb Yeah, I'm not experiencing any problems with it.  Just ideally shouldn't be displaying in red letters if it isn't a problem.

**Comment by superseb** [comment]
Ok thanks, we'll see what we can do about it.

**Comment by alena1108** [comment]
@ryanwalls @superseb the problem with the domain names list being too long will be that the user search domains (that come last), will never exercise. So although the warning is seemingly harmless (besides creating "noise" in the UI) , we'd have to fix it by removing the entries generated by rancher-dns code + some changes to rancher-dns microservice.

**Comment by moshe0076** [comment]
I get this message on Rancher 1.6.7 running k8s:v1.7.2-rancher7 when a new pod is launched.

**Comment by kinarashah** [comment]
A new label "io.rancher.container.dns.priority" has been temporarily introduced to solve this issue. Rancher internal search domains won't be added to kubelet if this label is set to None. 

In order to actually remove domains generated by rancher code, we'll introduce "serviceUUID.discover.internal" for all internal domains. This requires a lot of changes, and will be introduced only in versions > 1.6.x.

**Comment by kinarashah** [comment]
To test, 
a) use the [k8s catalog template](https://github.com/rancher/rancher-catalog/pull/858) which starts kubelet with this label set. 
b) login to kubelet and see no rancher specific domains are set. 
c) create a simple replication controller, and see that the POD's /etc/resolv.conf doesn't have any rancher specific search domains set. 
d) verify that the skydns resolution still works fine.

**Comment by kinarashah** [comment]
@sangeethah 

To test, 
1) start any regular user service by creating the service with label "io.rancher.container.dns.priority" : "None". Check its /etc/resolv.conf to confirm it doesn't have any rancher specific domains. 

2) In kubernetes, 
a) use the [k8s catalog template](https://github.com/rancher/rancher-catalog/pull/862) which starts kubelet with this label set. 
b) login to kubelet and see no rancher specific domains are set. 
c) create a simple replication controller, and see that the POD's /etc/resolv.conf doesn't have any rancher specific search domains set. 
d) verify that the skydns resolution still works fine.

**Comment by galal-hussein** [comment]
Rancher server: v1.6-development

I tested with normal services and in kubernetes, i don't see the problem anymore, and resolve.conf doesn't have rancher specific search domain set, skydns resolution still works just fine.

**Comment by moshe0076** [comment]
Hi,
I have Rancher 1.6.9 and agent v1.2.6 (Upgraded from 1.6.7). I have set the label 
io.rancher.container.dns.priority = None on the kubelets.

I can still see the error when creating new pods:

Search Line limits were exceeded, some dns names have been omitted, the applied search line is: ci-cd-infra.svc.cluster.local svc.cluster.local cluster.local kubelet.kubernetes.rancher.internal kubernetes.rancher.internal rancher.internal
Error syncing pod.

In /etc/resolve.conf I can see:

nameserver 10.43.0.10
search ci-cd-infra.svc.cluster.local svc.cluster.local cluster.local kubelet.kub
ernetes.rancher.internal kubernetes.rancher.internal rancher.internal
options ndots:5

Thanks!

Moshe

**Comment by deniseschannon** [comment]
@moshe0076 This has not been released and will be in the next Rancher and k8s release.

**Comment by moshe0076** [comment]
@deniseschannon - Thanks!
Do you have any ETA?

**Comment by arunkjn** [comment]
Hi, Any update on this or any workarounds? Does using `io.rancher.container.dns.priority = None` annotation on the service resolve this?

**Comment by vincent99** [comment]
The search length thing is not an _error_, it's a _warning_ that has no actual consequence and does not prevent things from running/working.  If your pods are not creating you have a different problem unrelated to this.

**Comment by arunkjn** [comment]
ok cool. One of my pods was not scheduling and giving this ~~error~~ warning. I will investigate what else could be the issue

**Comment by superseb** [comment]
This is fixed in v1.6.11. You should be able to run a `kubectl describe pod` to see what the problem is, or use the Kubernetes Dashboard.

**Comment by obriensystems** [comment]
Running 1.6.14 - reverifying
